test/controllers/TransferAgreementControllerSpec.scala

```diff
@@ -182,6 +169,35 @@ class TransferAgreementControllerSpec extends FrontEndTestHelper {
       contentAsString(transferAgreementSubmit) must include("govuk-error-message")
       contentAsString(transferAgreementSubmit) must include("error")
     }
+
+    "render the transfer agreement 'agreed' page with an authenticated user if consignment status is 'Completed'" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val controller = new TransferAgreementController(getAuthorisedSecurityComponents,
+        new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration, langs)
+
+      val client = new GraphQLConfiguration(app.configuration).getClient[gcs.Data, gcs.Variables]()
+      val consignmentResponse = gcs.Data(Option(GetConsignment(CurrentStatus(Some("Completed"), None))))
+      val data: client.GraphqlData = client.GraphqlData(Some(consignmentResponse))
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
+      val transferAgreementPage = controller.transferAgreement(consignmentId)
+        .apply(FakeRequest(GET, "/consignment/c2efd3e6-6664-4582-8c28-dcf891f60e68/transfer-agreement").withCSRFToken)
+      val transferAgreementPageAsString = contentAsString(transferAgreementPage)
+
+      playStatus(transferAgreementPage) mustBe OK
+      contentType(transferAgreementPage) mustBe Some("text/html")
+      headers(transferAgreementPage) mustBe TreeMap("Cache-Control" -> "no-store, must-revalidate")
+      transferAgreementPageAsString must include("Transfer agreement")
+      transferAgreementPageAsString must include("You have already confirmed all statements")
+      transferAgreementPageAsString must include("I confirm that the records are Public Records.")
+      transferAgreementPageAsString must include("I confirm that the records are all Crown Copyright.")
+      transferAgreementPageAsString must include("I confirm that the records are all in English.")
+      transferAgreementPageAsString must include("I confirm that the Departmental Records Officer (DRO) has signed off on the appraisal and selection")
+      transferAgreementPageAsString must include("I confirm that the Departmental Records Officer (DRO) has signed off on the sensitivity review.")
+      transferAgreementPageAsString must include("I confirm that all records are open and no Freedom of Information (FOI) exemptions apply to these records.")
```

Comment:
Could you add some checks to make sure that it's not possible to submit the form again?

Reply:
Is this the line that addresses this comment?

```
transferAgreementPageAsString must include(s"""<form action="/consignment/${consignmentId}/upload" method="GET" novalidate="">""")
```

I wonder if it would be better to test that there is no form submission button, since you're trying to demonstrate that there's no way to submit the form?

Reply:
I guess question is related to the question I answered below, should I add some logic do display either a submit button or a link button depending on if the user has submitted or not?